### ISSUE TITLE
Add [VB] coverage formats

### ISSUE THREAD

**Issue by stevepeak** [codecov/support#255]
- [x] add server side parser
- [x] add known file names to upload patterns with [codecov-python][1] uploader
- [ ] create example repository

[1]: https://github.com/codecov/codecov-python

**Comment by hebinhuang** [comment]
Updates for the offline conversation with @stevepeak. Thanks @stevepeak to take care the issue.

Currently, codecov does not supports the coverage outputs that generated by Visual Studio CodeCoverage.exe that comes with Visual Studio 2012/2013(Premium and Ultimate)/2015 (Professional and enterprise). By default CodeCoverage.exe creates a *.coverage file, then the file can be converted to *.xml format.

To get the *.xml file I use the following command:
"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Team Tools\Dynamic Code Coverage Tools\CodeCoverage.exe" collect /output:rio.coverage “PATH_OF_YOUR_EXECUTABLE_OR_DLL”
"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Team Tools\Dynamic Code Coverage Tools\CodeCoverage.exe" analyze /output:rioresults.xml rio.coverage

The attached file is the xml output for coverage:
[rioresults.zip](https://github.com/codecov/support/files/360372/rioresults.zip)

**Comment by stevepeak** [comment]
@hebinhuang 👍 perfect! Should have this finished by for tomorrow

**Comment by stevepeak** [comment]
@hebinhuang server side processing ready :) We can start uploading reports today.

---

I'm aware that the [bash uploader][1] will not work on Window. Our [python uploader][2] is the best option for now.

```
pip install --user codecov
codecov -t project_token
```
> any path at `*.coverage` will be found automatically.

[1]: https://github.com/codecov/codecov-bash
[2]: https://github.com/codecov/codecov-python

**Comment by hebinhuang** [comment]
@stevepeak, thanks very much for the work.

I have tried upload the file that attached before. It only can show the coverage for C#. The coverage xml also contains coverage for C++. But I did not see it.

https://codecov.io/gh/hebinhuang/Mobius/tree/06491e0e5873a280349c4a649af0cbca3f2f4e9c

![shotcutofcoverage](https://cloud.githubusercontent.com/assets/17152782/16848611/690a0f00-49ab-11e6-9d71-eb2937c5c32f.png)

**Comment by stevepeak** [comment]
@hebinhuang previously was not looping through modules. I fixed that up and deployed. Ready for another commit.

**Comment by hebinhuang** [comment]
@stevepeak, It looks like Visual Studio has two different coverage xml formats.
-  One is the format I mentioned above that converted by codecoverage.exe.
-  Another is converted by Visual Studio UI or a customer tool that use Microsoft.VisualStudio.Coverage.Analysis.dll.

You can refer to https://github.com/danielpalme/ReportGenerator/wiki/Visual-Studio-Coverage-Tools for some information.

The attached contains the two formats:
[NewCodeCoverage.zip](https://github.com/codecov/support/files/364466/NewCodeCoverage.zip)

**Comment by stevepeak** [comment]
Cool :) So the file at NewCodeCoverage-codecoverageexe-output.xml should be working. I'll need to review the other file type.

**Comment by hebinhuang** [comment]
@stevepeak, I have tried again for NewCodeCoverage-codecoverageexe-output.xml.

It still can see only one module. Not sure whether your fix is ready in online. 

https://codecov.io/gh/hebinhuang/Mobius/tree/970f853f8b7674b1e8349a7aad86a427389b3a10

**Comment by stevepeak** [comment]
Hey @hebinhuang planning on deploying this week. I'll return here to have you test it out. Thank you 👍

**Comment by stevepeak** [comment]
@hebinhuang deploying today :) Just checking on something: is the second report's first node always `<CoverageDSPriv>`?

**Comment by hebinhuang** [comment]
Thanks @stevepeak :)

Correct, the second report's first node always <CoverageDSPriv>

**Comment by stevepeak** [comment]
Cool! This is deployed. Can you give it a spin  👍 Thanks

**Comment by hebinhuang** [comment]
@stevepeak, I just tried for the fix of the first reports. it looks it still can  parse only one module.

http://codecov.io/github/hebinhuang/Mobius/commit/1115ad45c45f6313ec05a8c05c4bf96c413db751

**Comment by hebinhuang** [comment]
@stevepeak, I have tried for the first report format. It works for multiple modules now.
https://codecov.io/gh/hebinhuang/Mobius/tree/1115ad45c45f6313ec05a8c05c4bf96c413db751

But for the second report format, it seems the coverage number is not correct - only has 16.06%
https://codecov.io/gh/hebinhuang/Mobius/tree/53b8a8743f040cf78c250f226159651e56107e1c

**Comment by stevepeak** [comment]
@hebinhuang 
TLDR;
1115ad45 contains `vb1` and `vb2

I reviewed the reports and found they are accurate with the 16.06%. I did notice that 
the reports for 53b8a87 are both the same format (lets call it `vb2`). I'm not seeing any `vb1` report data.

The commit at 1115ad45 contains `vb1` reports, which express the higher coverage.

**Comment by hebinhuang** [comment]
@stevepeak, I submitted another reports for vb1. I only see one module results from there. Do I have to wait for a while to show the exact reports?

https://codecov.io/gh/hebinhuang/Mobius/tree/d02d932ef9990b4eae93e33e384150d57ad6e9fd

**Comment by stevepeak** [comment]
@hebinhuang should be immediate. I'll review the uploads and see if there are any issues. Thank you for helping me out!

**Comment by stevepeak** [comment]
@hebinhuang the [builds](https://codecov.io/gh/hebinhuang/Mobius/commit/d02d932ef9990b4eae93e33e384150d57ad6e9fd/builds) show only one report this time. [This report](https://codecov.io/github/archive/v4/raw/2016-07-21/D86D7DE6943D7B4DF0671988BAFDF5F2/d02d932ef9990b4eae93e33e384150d57ad6e9fd/0eb87f02-1c33-4536-9fe8-c19368947dc2.txt) only has content from `C:\projects\mobius-01vnv\csharp\Adapter\Microsoft.Spark.CSharp/*`.

Am I missing something?

**Comment by hebinhuang** [comment]
@stevepeak does all builds contains two reports? It looks like [this report](https://codecov.s3.amazonaws.com/v4/raw/2016-07-21/D86D7DE6943D7B4DF0671988BAFDF5F2/d02d932ef9990b4eae93e33e384150d57ad6e9fd/0eb87f02-1c33-4536-9fe8-c19368947dc2.txt) contains 3 modules. and all source files.

Not sure why it only shows the content from CSharp*

**Comment by hebinhuang** [comment]
[This builds](https://codecov.io/gh/hebinhuang/Mobius/commit/d9691e2578e6e3fa1e13a38310e1938a9a83bc61/builds) for vb2 format also show only one report. But, it can show all modules.

**Comment by stevepeak** [comment]
Let's jump in a chat https://gitter.im/codecov/support

**Comment by hebinhuang** [comment]
@stevepeak I have confirmed with two reports after deploying your fix. It works correctly for both vb1 and vb2.

VB2: https://codecov.io/gh/hebinhuang/Mobius/tree/e27aa20a52a1fb15966dd38f66f5e60ca45cfb97
VB1: https://codecov.io/gh/hebinhuang/Mobius/tree/98316fdb40bc0aa9f664174d7a820025fb1f08b2

I really appreciate your hard work and the support. 👍

**Comment by stevepeak** [comment]
👏 Awesomeness!

**Comment by stevepeak** [comment]
@hebinhuang https://github.com/danielpalme/ReportGenerator/issues/53
Thank you!
